src/libstd/io/mod.rs

```diff
@@ -322,6 +587,21 @@ pub trait Write {
     ///
     /// It is **not** considered an error if the entire buffer could not be
     /// written to this writer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::io::Write;
+    /// use std::fs::OpenOptions;
+    ///
+    /// # fn foo() -> std::io::Result<()> {
+    /// let mut buffer = try!(OpenOptions::new().write(true).open("foo.txt"));
+    ///
+    /// buffer.write(b"some bytes").unwrap();
+    /// buffer.flush().unwrap();
```

Comment:
before, you had said they were useless on vectors, but yeah, i guess that they theoretically don't need to be flushed here. i figure it's generally a good idea....